internal/node/client.go

```diff
@@ -50,40 +46,31 @@ func (client *client) GetMetricsURL() string {
 	return client.MetricsBaseURL.String()
 }
 
-// GetNodeMetrics retrieves polkadot metrics from prometheus server
-func (client *client) GetMetrics() (*Metrics, error) {
-	metricsURL, _ := client.MetricsBaseURL.Parse(metricsEndpoint)
-	resp, err := http.Get(metricsURL.String())
-	if err != nil {
-		return nil, fmt.Errorf("Metrics endpoint returned error: %v", err)
-	} else if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("Metrics endpoint returned invalid status code: %d", resp.StatusCode)
+// SendsRPCRequest sends rpc request to node rpc url and decodes result to v
+func (client *client) sendRPCRequest(method string, params []string, v interface{}) (*http.Response, error) {
```

Comment:
considering our use case, it would probably be better to use batch json rpc requests. Pls open issue for that but we would deal with that later